### PR TITLE
handle pm times when locale set to 12 hrs

### LINKED ISSUE #6102: encounter datetime bug when locale set to 12hrs
[reported via forum](https://community.open-emr.org/t/7-0-0-1-encounter-date-field-of-encounter-form-not-retaining-the-date/19257)

### PR DESCRIPTION
**Pull request by stephenwaite**
Fixes #6102

### CODE REVIEW

File: library/formatting.inc.php [hunk 1]
```diff
@@ -178,13 +178,17 @@ function DateToYYYYMMDD($DateValue)
...
 function TimeToHHMMSS($TimeValue)
 {
-    //For now, just return the $TimeValue, since input fields are not formatting time.
-    // This can be upgraded if decided to format input time fields.
-
     if (trim($TimeValue) == '') {
         return '';
     }
 
+    $is_pm = (stripos($TimeValue, 'PM') !== false);
```
Comment (robertdown) [inline library/formatting.inc.php:185]: Will this ignore lower case `pm`?

Comment (robertdown) [reply library/formatting.inc.php:185]: Ignore me.... it's `stripos`.... it's literally case insensitive

**Review by bradymiller — APPROVED** [review]

**Comment by bradymiller** [comment]
Just noted these issues after the approve.
1. Will seconds need to be managed also?
2. Will the corresponding js function, TimeToHHMMSS_js(), also need updating?

**Comment by stephenwaite** [comment]
when `oeFormatTime()` returns the global time display format for 12 hrs the datetimepicker ignores the `pm` and defaults the time to `am`
Screenshot from 2023-01-16 21-33-26
Screenshot from 2023-01-16 21-33-38

File: library/formatting_DateToYYYYMMDD_js.js.php [hunk 2]
```diff
@@ -32,8 +32,15 @@ function DateToYYYYMMDD_js(value){
...
-    // This can be upgraded if decided to format input time fields.
+    if (value.trim() == '') {
+        return '';
+    }
+
+    var is_pm = toUpperCase(value.trim()).indexOf('PM');
+    if (is_pm) {
+        let d = new Date("1970-01-01 " + value);
+        let value = d.setHours(d.getHours() + 12).toTimeString();
+    }
```
Comment (stephenwaite) [inline library/formatting_DateToYYYYMMDD_js.js.php:43]: haven't tested this :grimacing: 

looks like would have to run some cqm reports to do so

File: library/js/xl/jquery-datetimepicker-2-5-4.js.php [hunk 3]
```diff
@@ -92,7 +92,9 @@
             <?php } ?>
         <?php } else { ?>
             <?php if ($datetimepicker_formatInput) { ?>
-                format: '<?php echo DateFormatRead("jquery-datetimepicker"); ?> H:i',
+                format: '<?php echo DateFormatRead("jquery-datetimepicker"); ?> g:i a',
+                formatTime:'g:i a',
+                validateOnBlur: false,
```
Comment (stephenwaite) [inline library/js/xl/jquery-datetimepicker-2-5-4.js.php:97]: this at least prevents the pm time from converting to am and the `validateOnBlur()` prevents an hour prior time from being selected if clicking into the date field.

File: library/formatting_DateToYYYYMMDD_js.js.php (hunk 2 above)
Comment (stephenwaite) [reply library/formatting_DateToYYYYMMDD_js.js.php:43]: tested it a little bit to get it working for cqm report generation in Reports->Clinic->Automated Measures (AMC)

**Review by bradymiller — APPROVED** [review]

**Comment by bradymiller** [comment]
dance-bear

**Pull request merged**